docs/synapse/userguides/storm_adv_vars.rstorm

```diff
@@ -232,204 +316,268 @@ The variable name must be specified first, followed by the equals sign and the v
-
-.. storm-cli:: inet:fqdn=mail.mydomain.com $fqdn=$node.value() -> inet:dns:a +{ -> inet:ipv4 +:dns:rev=$fqdn }
+- **Tag timestamps:** Assign the times associated with Threat Group 20’s control of a malicious domain to the variable
+  ``$time``:
 
+.. storm-pre:: [inet:fqdn=evildomain.com +#cno.threat.t20.own=(2019/09/08,2021/09/08)]
+.. storm-cli:: inet:fqdn=evildomain.com $time=#cno.threat.t20.own $lib.print($time)
 
-*Assign a library function to a user-defined variable:*
+*Example*
 
-- Assign a value to the variable ``$mytag`` using a library function:
+- Find DNS A records for any subdomain associated with a Threat Group 20 FQDN (zone) during the time they
+  controlled the domain:
 
+.. storm-pre:: [ (inet:dns:a=(www.evildomain.com,1.2.3.4) .seen=(2020/07/12,2020/12/13)) (inet:dns:a=(smtp.evildomain.com,5.6.7.8) .seen=(2020/04/04,2020/08/02)) (inet:dns:a=(evildomain.com,12.13.14.15) .seen=(2021/12/22,2022/12/22))]
+.. storm-cli:: inet:fqdn#cno.threat.t20.own $time=#cno.threat.t20.own -> inet:fqdn:zone -> inet:dns:a +.seen@=$time
 
-.. storm-cli:: $mytag = $lib.str.format("cno.mal.sofacy") $lib.print($mytag)
 
+**Library Functions**
 
-- Assign a value to the variable ``$mytag`` using a library function (example 2):
+Storm types (Storm objects) and Storm libraries allow you to inspect, edit, and otherwise work with data in
+Synapse in various ways. You can assign a variable based on the output of a method or library.
 
-.. storm-pre:: [ file:bytes=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 +#code.fam.sofacy ]
-.. storm-cli:: file:bytes=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 for $tag in $node.tags(code.fam.*) { $malfam=$tag.split(".").index(2) $mytag=$lib.str.format("cno.mal.{malfam}", malfam=$malfam) $lib.print($mytag) } | spin
-    
+A full discussion of this topic is outside of the scope of this user guide. See :ref:`stormtypes_index` for
+additional details.
 
 
-The above example leverages:
+- Assign the current time to the variable ``$now`` using ``$lib.time.now()``:
 
- - three variables (``$tag``, ``$malfam``, and ``$mytag``);
- - the :ref:`meth-node-tags` method;
- - the ``$lib.split()``, ``$lib.index()``, and ``$lib.str.format()`` library functions; as well as
- - a "for" loop.
+.. storm-cli:: $now=$lib.time.now() $lib.print($now)
 
+- Convert an epoch milliseconds integer into a human-readable date/time string using ``$lib.str.format()``:
 
+.. storm-cli:: $now=$lib.time.now() $time=$lib.time.format($now, '%Y/%m/%d %H:%M:%S') $lib.print($time)
 
-- If a file is tagged as part of a malicious code (malware) family, then also tag the file to indicate it is part of that malware's ecosystem:
+**Expressions**
 
+You can assign the value of a variable based on the computed value of an expression. To denote an expression
+to be evaluated, enclose the expression in parentheses preceded by a dollar sign (``$( <expression> )``).
```

Comment:
you don't actually need the `$` to do this now. the only place where expressions require a `$` is in a filter (to differentiate from a compound filter)

Reply:
also 'assign the value of a variable' doesn't sound quite right to me, 'assign a value to a variable' seems better